java/src/test/java/com/saucelabs/simplesauce/SauceOptionsTest.java

```diff
@@ -1,50 +1,120 @@
 import org.openqa.selenium.MutableCapabilities;
 import org.openqa.selenium.PageLoadStrategy;
 import org.openqa.selenium.chrome.ChromeOptions;
 import org.openqa.selenium.edge.EdgeOptions;
 import org.openqa.selenium.firefox.FirefoxOptions;
 import org.openqa.selenium.ie.InternetExplorerOptions;
-import org.openqa.selenium.remote.BrowserType;
 import org.openqa.selenium.safari.SafariOptions;
 
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 import static org.junit.Assert.assertEquals;
 import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.spy;
 import static org.openqa.selenium.UnexpectedAlertBehaviour.DISMISS;
 
-public class SauceOptionsTest extends BaseConfigurationTest{
+public class SauceOptionsTest {
+    protected SauceOptions sauceOptions = new SauceOptions();
+
+    @Rule
+    public MockitoRule initRule = MockitoJUnit.rule();
+
     @Test
-    public void usesLatestChromeWindowsVersions() {
-        assertEquals(BrowserType.CHROME, sauceOptions.getBrowserName());
-        assertEquals("Windows 10", sauceOptions.getPlatformName());
+    public void usesLatestChromeWindowsVersionsByDefault() {
+        assertEquals(Options.Browser.CHROME, sauceOptions.getBrowserName());
+        assertEquals(Options.Platform.WINDOWS_10, sauceOptions.getPlatformName());
         assertEquals("latest", sauceOptions.getBrowserVersion());
     }
 
     @Test
     public void updatesBrowserBrowserVersionPlatformVersionValues() {
-        sauceOptions.setBrowserName(BrowserType.FIREFOX);
-        sauceOptions.setPlatformName(Platforms.MAC_OS_HIGH_SIERRA.getOsVersion());
+        sauceOptions.setBrowserName(Options.Browser.FIREFOX);
+        sauceOptions.setPlatformName(Options.Platform.MAC_HIGH_SIERRA);
         sauceOptions.setBrowserVersion("68");
 
-        assertEquals(BrowserType.FIREFOX, sauceOptions.getBrowserName());
+        assertEquals(Options.Browser.FIREFOX, sauceOptions.getBrowserName());
         assertEquals("68", sauceOptions.getBrowserVersion());
-        assertEquals("macOS 10.13", sauceOptions.getPlatformName());
+        assertEquals(Options.Platform.MAC_HIGH_SIERRA, sauceOptions.getPlatformName());
+    }
+
+    @Test
+    public void fluentPatternWorks() {
+        sauceOptions.setBrowserName(Options.Browser.FIREFOX)
+                .setBrowserVersion("68")
+                .setPlatformName(Options.Platform.MAC_HIGH_SIERRA);
+
+        assertEquals(Options.Browser.FIREFOX, sauceOptions.getBrowserName());
+        assertEquals("68", sauceOptions.getBrowserVersion());
+        assertEquals(Options.Platform.MAC_HIGH_SIERRA, sauceOptions.getPlatformName());
     }
 
     @Test
-    @Ignore("Not Implemented Yet")
     public void acceptsOtherW3CValues() {
+        sauceOptions.setPageLoadStrategy(Options.PageLoadStrategy.EAGER);
+        sauceOptions.setAcceptInsecureCerts(true);
+        Map<Options.Timeouts, Integer> timeouts = new HashMap<>();
```

Comment:
I think I previously asked about this and maybe forgot, but why did we decide to go this route versus this?

```
sauceOptions.timeouts.setImplicit(1);
sauceOptions.timeouts.pageLoad(100);
sauceOptions.timeouts.script(10);
```

Reply:
I'm cool with it, I just am not sure what the implementation for it would look like.

Reply:
Oh I see it would need to be typed to another class, we couldn't use it as both a Map and an Timeouts object...  hmmm. The problem is that the reflection implementation of `toCapabilities()` requires it to be a map... Maybe I can override the setter for it...

Reply:
What do you think of this implementation? 
https://github.com/saucelabs/simple_sauce/commit/5f19f8225afcbd08fa593d7d289ee7042d991934

Reply:
Nevermind, I just added it to this PR.  Singlular `timeout` and `setX(val)` syntax.